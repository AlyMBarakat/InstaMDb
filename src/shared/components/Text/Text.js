import React from "react";
import { Text, StyleSheet } from "react-native";

const CustomText = ({ size, color, style, children, regular, light, bold, ...rest }) => {
    // Determine font weight
    let fontFamily = "Rubik-Medium"; // Default
    if (regular) fontFamily = "Rubik-Regular";
    if (light) fontFamily = "Rubik-Light";
    if (bold) fontFamily = "Rubik-Bold";

    //Determine font size
    let fontSize = 16;
    if (size) fontSize = size;

    //Determine font color
    let textColor = "white";
    if (color) textColor = color;

    return (
        <Text
            {...rest}
            style={[{ fontFamily, fontSize, color: textColor }, style]}
        >{children}</Text>
    );
}

export default CustomText;