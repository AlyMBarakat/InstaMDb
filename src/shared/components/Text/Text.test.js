import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../Text';

test('renders font weight, size and color correctly', () => {
    const tree = renderer.create(<>
        <Text light size={23} color="white">testing</Text>
        <Text bold size={19} color="black">testing</Text>
        <Text regular size={80} color="#2d2d2d">testing</Text>
    </>).toJSON();
    expect(tree).toMatchSnapshot();
});

