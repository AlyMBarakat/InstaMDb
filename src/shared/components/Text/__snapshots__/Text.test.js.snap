// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders font weight and size correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Light",
          "fontSize": 23,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Bold",
          "fontSize": 19,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Regular",
          "fontSize": 80,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
]
`;

exports[`renders font weight correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Light",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Bold",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Regular",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
]
`;

exports[`renders font weight, size and color correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Rubik-Light",
          "fontSize": 23,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Rubik-Bold",
          "fontSize": 19,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#2d2d2d",
          "fontFamily": "Rubik-Regular",
          "fontSize": 80,
        },
        undefined,
      ]
    }
  >
    testing
  </Text>,
]
`;
